internal/metamorph/store/postgresql/postgres_test.go

```diff
@@ -555,15 +558,114 @@ func TestPostgresDB(t *testing.T) {
 			MerklePath:      "",
 		}}
 
-		_, err := postgresDB.UpdateMined(ctx, txBlocks)
+		dataBeforeUpdate, err := postgresDB.Get(ctx, unminedHash[:])
+		require.NoError(t, err)
+
+		_, err = postgresDB.UpdateMined(ctx, txBlocks)
 		require.NoError(t, err)
 
 		dataReturned, err := postgresDB.Get(ctx, unminedHash[:])
 		require.NoError(t, err)
+		unmined.BlockHeight = 0
+		unmined.BlockHash = nil
+		unmined.StatusHistory = append(unmined.StatusHistory, &store.StoreStatus{
+			Status:    dataBeforeUpdate.Status,
+			Timestamp: dataBeforeUpdate.LastModified,
+		})
 		unmined.Status = metamorph_api.Status_MINED
+		unmined.LastModified = dataReturned.LastModified
+
+		require.Equal(t, &unmined, dataReturned)
+	})
+
+	t.Run("update mined - all possible updates", func(t *testing.T) {
+		defer pruneTables(t, postgresDB.db)
+
+		unmined := *unminedData
+		err = postgresDB.Set(ctx, &unmined)
+		require.NoError(t, err)
+
+		// First update - UpdateStatusBulk
+		updates := []store.UpdateStatus{
+			{
+				Hash:   *unminedHash,
+				Status: metamorph_api.Status_ACCEPTED_BY_NETWORK,
+			},
+		}
+
+		dataBeforeUpdate, err := postgresDB.Get(ctx, unminedHash[:])
+		require.NoError(t, err)
+
+		statusUpdates, err := postgresDB.UpdateStatusBulk(ctx, updates)
+		require.NoError(t, err)
+		require.Len(t, statusUpdates, 1)
+
+		updatedTx, err := postgresDB.Get(ctx, unminedHash[:])
+		require.NoError(t, err)
+
 		unmined.BlockHeight = 0
 		unmined.BlockHash = nil
-		require.Equal(t, dataReturned, &unmined)
+		unmined.StatusHistory = append(unmined.StatusHistory, &store.StoreStatus{
+			Status:    dataBeforeUpdate.Status,
+			Timestamp: dataBeforeUpdate.LastModified,
+		})
+		unmined.Status = metamorph_api.Status_ACCEPTED_BY_NETWORK
+		unmined.LastModified = updatedTx.LastModified
```

Comment:
I think here we can set unmined.LastModified = now

Reply:
Done